src/IO/HDF5/HDF5IOHandler.cpp

```diff
@@ -2210,6 +2288,42 @@ void HDF5IOHandlerImpl::readAttribute(
             status = H5Aread(attr_id, attr_type, vcld.data());
             a = Attribute(vcld);
         }
+        else if (H5Tequal(attr_type, m_H5T_CLONG_DOUBLE_80_LE))
+        {
+            // worst case, sizeof(long double) is only 8, so allocate enough
+            // memory to fit 16 bytes per member
+            auto *tmpBuffer =
+                static_cast<long double *>(malloc(16 * 2 * dims[0]));
```

Comment:
In C++, I would use `new` and `delete` over `malloc` and `free`.

Reply:
This will need a `reinterpret_cast` then though, as you can't `new` a void pointer. But I'll push a commit.

Reply:
We found today: can be `new long double` :sweat_smile:

Reply:
It turns out, we cannot 🙃 
This branch is active when `sizeof(long double) == 8`, but the dataset on disk has 16-byte doubles. There is no way to allocate a correctly-sized buffer using native float types, so `new char[]` it is.